persistence/src/main/java/com/epam/esm/repository/AbstractRepository.java

```diff
@@ -0,0 +1,68 @@
+package com.epam.esm.repository;
+
+import com.epam.esm.model.Entity;
+import com.epam.esm.repository.specification.Specification;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Optional;
+
+public abstract class AbstractRepository<T extends Entity> implements MainRepository<T> {
+    private static final String DELETE_BY_ID_QUERY = "DELETE FROM %s WHERE id = ?";
+    private static final String ID_ATTRIBUTE = "id";
+    private static final String EXISTS_QUERY = "SELECT EXISTS(%s) as ex";
+    protected final JdbcTemplate jdbcTemplate;
+    protected final RowMapper<T> mapper;
+
+    protected AbstractRepository(JdbcTemplate jdbcTemplate, RowMapper<T> mapper) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.mapper = mapper;
+    }
+
+    protected int insert(String query, Object...params) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        PreparedStatementCreator psc = connection -> {
```

Comment:
Try to avoid naming like psc or ps. Better to use more clear names, for example statementCreator and statement.